src/training/stylegan.py

```diff
@@ -0,0 +1,100 @@
+        self.total_epochs = total_epochs
+        self.logger = logger
+        self.storage = storage
+        self.ocr_loss = ocr.OCRLoss().to(device)
+        self.perceptual_loss = perceptual_loss.to(device)
+        self.coef_ocr = coef_ocr_loss
+        self.coef_perceptual = coef_perceptual_loss
+        model_ft = models.resnet18(pretrained=True)
+        self.style_embedder   = torch.nn.Sequential(*list(model_ft.children())[:-1]).to(device)
+        self.content_embedder = torch.nn.Sequential(*list(model_ft.children())[:-2]).to(device)
+
+    
+    def train(self):
+        logger.info('Start training')
+        self.model.train()
+
+        for style_batch, content_batch, label_batch in self.train_dataloader:
+            style_batch = style_batch.to(self.device)
+            content_batch = content_batch.to(self.device)
+            style_embeds = self.style_embedder(style_batch)
+            content_embeds = self.content_embedder(content_batch)
+            self.optimizer.zero_grad()
+
+            res = self.model(content_embeds, style_embeds)
+            ocr_loss = self.ocr_loss(res, label_batch)
+            perceptual_loss = self.perceptual_loss(style_batch, res)
+            loss = self.coef_ocr * ocr_loss +  self.coef_perceptual * perceptual_loss
+            loss.backward()
+            self.optimizer.step()
+
+            self.logger.log_train(
+                losses={'ocr_loss': ocr_loss.item(), 'perceptual_loss': perceptual_loss.item(), 'full_loss': loss.item()},
+                images={'style': style_batch, 'content': content_batch, 'result': res}
+            )
+
+
+    def validate(self, epoch):
+        self.model.eval()
+
+        for style_batch, content_batch, label_batch in self.val_dataloader:
+            style_batch = style_batch.to(self.device)
+            content_batch = content_batch.to(self.device)
+            style_embeds = self.style_embedder(style_batch)
+            content_embeds = self.content_embedder(content_batch)
+
+            res = self.model(content_embeds, style_embeds)
+            ocr_loss = self.ocr_loss(res, label_batch)
+            perceptual_loss = self.perceptual_loss(style_batch, res)
+            loss = self.coef_ocr * ocr_loss +  self.coef_perceptual * perceptual_loss
+            
+            self.logger.log_val(
+                losses={'ocr_loss': ocr_loss.item(), 'perceptual_loss': perceptual_loss.item(), 'full_loss': loss.item()},
+                images={'style': style_batch, 'content': content_batch, 'result': res}
+            )
+
+        avg_losses, _ = self.logger.end_val()
+        self.storage.save(
+            epoch,
+            {'model': self.model, 'optimizer': self.optimizer, 'scheduler': self.scheduler},
+            avg_losses['full_loss']
+        )
+
+
+    def run(self):
+        for epoch in range(self.total_epochs):
+            self.train()
+            with torch.no_grad():
+              if (epoch + 3) % 10 == 0:
```

Comment:
If you are doing this because validation takes too long, maybe you could add `cut` options to `BaselineDataset` as we did it in `ColorDataset`: [link](https://github.com/grenlayk/text-deep-fake/blob/dc0be62766452808f141ca039c89b880aa4e833d/src/data/color.py#L17)